server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -826,6 +832,10 @@ func (e *EventChannel) processSingleEvent(event *inpb.InvocationEvent, iid strin
 	e.targetTracker.TrackTargetsForEvent(e.ctx, event.BuildEvent)
 	e.statusReporter.ReportStatusForEvent(e.ctx, event.BuildEvent)
 
+	if err := e.collectAPIFacets(iid, event.BuildEvent); err != nil {
```

Comment:
Should we hate this behind the cache enabled flag so we can turn it off easily if it causes problems?

Reply:
Done.